.github/workflows/main.yml

```diff
@@ -0,0 +1,43 @@
+#main.yml
+
+name: Main
+on: pull_request
+
+jobs:
+  checks:
+    runs-on: macOS-latest
+    steps:
+
+      - name: Project Checkout
+        uses: actions/checkout@v2.3.4
+
+      - name: Compute key for CI cache
+        run: ./.github/workflows/compute-ci-cache-key.sh key.txt
+
+      - name: Setup cache
+        uses: actions/cache@v2.1.3
+        continue-on-error: true
+        with:
+          path: ~/.gradle/caches
+          key: ${{ runner.os }}-gradle-${{ hashFiles('key.txt') }}
+          restore-keys: ${{ runner.os }}-gradle-
+
+      - name: Setup JDK 1.8
+        uses: actions/setup-java@v1.4.3
+        with:
+          java-version: 1.8
+
+      - name: Clean project
+        run: ./gradlew clean
+
+      - name: Build project
+        run: ./gradlew build --stacktrace
+
+      - name: Assemble Test APK
+        run: ./gradlew assembleAndroidTest --stacktrace
+
+      - name: Espresso Checks
+        uses: reactivecircus/android-emulator-runner@v2.13.0
+        with:
+          api-level: 23
```

Comment:
On Travis the emulator is being running for SDK level `16`; not sure if this is a hard requirement.

Also : not that this job can run in a `build-matrix` fashion, targeting multiple Android SDK levels for Espresso Checks; maybe this would be a nice upcoming PR :)

Reply:
There is this specific line

https://github.com/square/leakcanary/blob/main/.travis.yml#L20

that warns to avoid `clean` and `build` running in parallel. I just followed the same. 

On top of that, the very first check failed with [this error](https://github.com/square/leakcanary/runs/1523047528?check_suite_focus=true), not sure if it is related to a know issue / race condition.

Reply:
That line doesn't explain why the clean is there at all in travis, although it could be that travis was leaking files between builds, and this was the workaround. Shouldn't be needed here.